Automattic/wp-calypso
# Segmentation Survey: Fix save answer and other small improvements

## Proposed Changes

* Fix issue when saving answers for a question
  * It was skipping to save for all questions after the previous PR. The previous approach was considering the first parameter to be a skip boolean, but the onClick was sending the click event as the first param.
  * I rewrote the skip and nextPage functions, removing the skip param dependency
* Move useSurveyAnswersQuery to client/data
* Add animation to checkbox inputs
* Update save answers request URL (it somehow got lost in the previous PR)

## Testing Instructions

* Apply this PR to your local
* Go to http://calypso.localhost:3000/setup/entrepreneur/start?flags=ecommerce-segmentation-survey
* Select a checkbox input and verify if the animation works as expected (should be consistent with the radio input animation)
* Select some options and check if the answers are correctly saved (see Network tab)
* Skip one of the questions. In this case, the answers should not be saved
* Check for regressions

## Pre-merge Checklist
